crates/analyzer/src/traversal/expressions.rs

```diff
@@ -705,18 +705,34 @@ fn expr_unary_operation(
 
         return match op.kind {
             fe::UnaryOperator::USub => {
-                let int_type = expected_type.as_int().unwrap_or(Integer::I256);
+                let expected_int_type = expected_type.as_int().unwrap_or(Integer::I256);
                 match operand_attributes.typ {
-                    Type::Base(Base::Numeric(_)) => {
+                    Type::Base(Base::Numeric(operand_type)) => {
                         if let fe::Expr::Num(num_str) = &operand.kind {
                             let num = -to_bigint(num_str);
-                            validate_numeric_literal_fits_type(context, num, exp.span, int_type);
+                            validate_numeric_literal_fits_type(
+                                context,
+                                num,
+                                exp.span,
+                                expected_int_type,
+                            );
+                        }
+
+                        if !(operand_type.is_signed() || expected_int_type.is_signed()) {
```

Comment:
The following code still isn't rejected though even though it should.

```
contract Foo:
    pub fn bar(self) -> u32:
        let signed: i32 = 1
        return -signed
```
I haven't dug into it but I think it would be worth to address this as well.

Reply:
>signed is i32 => -signed could be unsigned number(u32 in this case). => We shouldn't throw error.

Yes, I thought so, too for a brief moment but I checked Rust and they would not let you do this and so I think we shouldn't either.


```
fn foo() -> u32 {
    let signed: i32 = -1;
    return -signed
}
```

Even though `-` applied to `-1` would make it `1` and would fit `u32` Rust will still reject this and I think its for a good reason.

Reply:
> ```rust
> fn foo() -> u32 {
>     let signed: i32 = -1;
>     return -signed
> }
> ```

I worry how we can casting type of expression. Code below will throw error when compile. 

```python
contract Foo:
    pub fn bar() -> u32:
        let a: i32 = -1
        return u32(-a)
```

We can do somthing like below to casting type, but I don't like it.

```python
contract Foo:
    pub fn bar() -> u32:
        let a: i32 = -1
        let b: i32 = -a
        return u32(b)
```